drivers/gpio/gpio_ite_it8801.c

```diff
@@ -0,0 +1,486 @@
+	struct gpio_driver_config common;
+	/* IT8801 controller dev */
+	const struct device *mfd;
+	/* I2C device for the MFD parent */
+	const struct i2c_dt_spec i2c_dev;
+	/* GPIO input pin status register */
+	uint8_t reg_ipsr;
+	/* GPIO set output value register */
+	uint8_t reg_sovr;
+	/* GPIO control register */
+	uint8_t reg_gpcr;
+	/* GPIO interrupt status register */
+	uint8_t reg_gpisr;
+	/* GPIO interrupt enable register */
+	uint8_t reg_gpier;
+	uint8_t pin_mask;
+};
+
+struct gpio_it8801_data {
+	struct gpio_driver_data common;
+	const struct device *gpio_dev;
+	sys_slist_t callbacks;
+	uint8_t level_high_trig;
+	uint8_t level_low_trig;
+};
+
+static int ioex_check_is_not_valid(const struct device *dev, gpio_pin_t pin)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret = 0;
+
+	if (BIT(pin) & ~(config->pin_mask)) {
+		LOG_ERR("GPIO  port%d-%d is not support", config->reg_ipsr, pin);
+		return -ENOTSUP;
+	}
+
+	return ret;
+}
+
+static int gpio_it8801_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret;
+	uint8_t reg_gpcr = config->reg_gpcr + pin;
+	uint8_t mask = BIT(pin);
+	uint8_t new_value = 0, control;
+
+	/* Don't support "open source" mode */
+	if (((flags & GPIO_SINGLE_ENDED) != 0) && ((flags & GPIO_LINE_OPEN_DRAIN) == 0)) {
+		return -ENOTSUP;
+	}
+
+	if (ioex_check_is_not_valid(dev, pin)) {
+		return -ENOTSUP;
+	}
+
+	ret = i2c_reg_read_byte_dt(&config->i2c_dev, reg_gpcr, &control);
+	if (ret) {
+		LOG_ERR("Failed to read control value (ret %d)", ret);
+		return ret;
+	}
+
+	if (flags == GPIO_DISCONNECTED) {
+		control &= ~(IT8801_GPIODIR | IT8801_GPIOPDE | IT8801_GPIOPUE);
+
+		goto write_and_return;
+	}
+
+	/* If output, set level before changing type to an output. */
+	if (flags & GPIO_OUTPUT) {
+		if (flags & GPIO_OUTPUT_INIT_HIGH) {
+			new_value = mask;
+		} else if (flags & GPIO_OUTPUT_INIT_LOW) {
+			new_value = 0;
+		}
+		ret = i2c_reg_update_byte_dt(&config->i2c_dev, config->reg_sovr, mask, new_value);
+		if (ret) {
+			LOG_ERR("Failed to set output value (ret %d)", ret);
+			return ret;
+		}
+		/* Set output */
+		control |= IT8801_GPIODIR;
+		/* Select open drain 0:push-pull 1:open-drain */
+		if (flags & GPIO_OPEN_DRAIN) {
+			control |= IT8801_GPIOIOT_OD;
+		} else {
+			control &= ~IT8801_GPIOIOT_OD;
+		}
+	} else {
+		/* Set input */
+		control &= ~IT8801_GPIODIR;
+	}
+
+	/* Handle pullup / pulldown */
+	if (flags & GPIO_PULL_UP) {
+		control = (control | IT8801_GPIOPUE) & ~IT8801_GPIOPDE;
+	} else if (flags & GPIO_PULL_DOWN) {
+		control = (control | IT8801_GPIOPDE) & ~IT8801_GPIOPUE;
+	} else {
+		/* No pull up/down */
+		control &= ~(IT8801_GPIOPUE | IT8801_GPIOPDE);
+	}
+
+write_and_return:
+	/* Set GPIO control */
+	ret = i2c_reg_write_byte_dt(&config->i2c_dev, reg_gpcr, control);
+	if (ret) {
+		LOG_ERR("Failed to set control value (ret %d)", ret);
+		return ret;
+	}
+
+	return ret;
```

Comment:
`return 0`, can't be anything else, removes any ambiguity

Reply:
Done.